scripts/publish.ts

```diff
@@ -10,40 +10,54 @@ import {
 import {getValueByFlag, hasFlag} from './shared/argv.utils';
 
 const path = getValueByFlag<string>(`--path`, ``);
-const dryRun = hasFlag(`--dry-run`);
 
 (async function main(): Promise<void> {
     const packageJson = await import(resolve(path, `package.json`));
+    const versions: string[] = getAllVersions(packageJson.name);
 
     infoLog(`name: ${packageJson.name}`);
     infoLog(`version: ${packageJson.version}`);
 
-    if (versionExists()) {
+    if (versions.includes(packageJson.version)) {
         errorLog(`${packageJson.name}@${packageJson.version} is already published`);
 
         return;
     }
 
-    const command = `npm publish ${path} ${makeAdditionalFlags()} --access public`;
+    const dry = hasFlag(`--dry-run`) ? `--dry-run` : ``;
+    const tag = makeTag(packageJson.version, versions);
+    const command = `npm publish ${path} ${tag} ${dry} --access public`;
 
     processLog(command);
     execSync(command, {stdio: `inherit`, encoding: `utf8`});
     successLog(`+${packageJson.name}@${packageJson.version} is published successfully`);
+})();
 
-    function versionExists(): boolean {
-        const versions: string[] = JSON.parse(
-            execSync(
-                `npm view ${packageJson.name} versions --json || echo "[]"`,
-            ).toString(),
-        );
+function getAllVersions(name: string): string[] {
+    const command = `npm view ${name} versions --json`;
 
-        return versions.includes(packageJson.version);
-    }
+    infoLog(`fetch: ${command}`);
+
+    return JSON.parse(execSync(`${command} || echo "[]"`).toString());
+}
+
+function makeTag(version: string, versions: string[]): string {
+    const currentMajor = parseInt(version);
+    const maxMajorVersion = getMaxMajorVersion(currentMajor, versions);
 
-    function makeAdditionalFlags(): string {
-        const tagNextFlag = packageJson?.version.includes(`rc`) ? `--tag next` : ``;
-        const dryRunFlag = dryRun ? `--dry-run` : ``;
+    return version.includes(`rc`)
+        ? `--tag next`
+        : maxMajorVersion > currentMajor
+        ? `--tag v${currentMajor}`
+        : ``;
```

Comment:
nit: we avoid ternary constructions like this. What about 
```
const tagFlag = maxMajorVersion > currentMajor ? `--tag v${currentMajor}` : ''
return version.includes(`rc`) ? '--tag next' : tagFlag
```

Reply:
fixed